docs/resources/ruleset.md

```diff
@@ -2,71 +2,71 @@
 
-The [Cloudflare Ruleset Engine](https://developers.cloudflare.com/firewall/cf-rulesets)
+The Cloudflare Ruleset Engine (https://developers.cloudflare.com/ruleset-engine/about/)
 allows you to create and deploy rules and rulesets.
 
-The engine syntax, inspired by the Wireshark Display Filter language, is the
-same syntax used in custom Firewall Rules. Cloudflare uses the Ruleset Engine
-in different products, allowing you to configure several products using the same
-basic syntax.
+Cloudflare uses the Ruleset Engine in different products, allowing
+you to configure several products using the same basic syntax.
 
 ## Example Usage
 
 ```terraform
 # Magic Transit
 resource "cloudflare_ruleset" "magic_transit_example" {
   account_id  = "f037e56e89293a057740de681ac9abbe"
-  name        = "account magic transit"
-  description = "example magic transit ruleset description"
-  kind        = "root"
+  name        = "My example Magic Transit ruleset"
+  description = "My example Magic Transit ruleset description"
   phase       = "magic_transit"
+  kind        = "root"
 
   rules {
-    action      = "allow"
-    expression  = "tcp.dstport in { 32768..65535 }"
+    ref         = "allow_tcp_ephemeral_ports"
     description = "Allow TCP Ephemeral Ports"
+    expression  = "tcp.dstport in { 32768..65535 }"
+    action      = "allow"
   }
 }
 
 # Zone-level WAF Managed Ruleset
 resource "cloudflare_ruleset" "zone_level_managed_waf" {
   zone_id     = "0da42c8d2132a9ddaf714f9e7c920711"
-  name        = "managed WAF"
-  description = "managed WAF ruleset description"
-  kind        = "zone"
+  name        = "My example managed WAF ruleset"
+  description = "My example managed WAF ruleset description"
   phase       = "http_request_firewall_managed"
+  kind        = "zone"
 
   rules {
-    action = "execute"
+    ref         = "execute_managed_ruleset"
+    description = "Execute Cloudflare Managed Ruleset on my zone-level phase entry point ruleset"
+    expression  = "(http.host eq \"example.host.com\")"
+    action      = "execute"
     action_parameters {
       id = "efb7b8c949ac4650a09736fc376e9aee"
     }
-    expression  = "(http.host eq \"example.host.com\")"
-    description = "Execute Cloudflare Managed Ruleset on my zone-level phase entry point ruleset"
-    enabled     = true
   }
 }
 
 # Zone-level WAF with tag-based overrides
 resource "cloudflare_ruleset" "zone_level_managed_waf_with_category_based_overrides" {
   zone_id     = "0da42c8d2132a9ddaf714f9e7c920711"
-  name        = "managed WAF with tag-based overrides"
-  description = "managed WAF with tag-based overrides ruleset description"
-  kind        = "zone"
+  name        = "My example managed WAF ruleset with tag-based overrides"
+  description = "My example managed WAF ruleset with tag-based overrides ruleset description"
   phase       = "http_request_firewall_managed"
+  kind        = "zone"
 
   rules {
-    action = "execute"
+    ref         = "execute_managed_ruleset"
```

Comment:
these new `ref` example look great. this provides a more complete set of option. thanks!